index.spec.ts

```diff
@@ -60,7 +63,20 @@ describe("Create a card", () => {
 			const submitButton = "#createCardForm #submitBtn"
 			await page.waitForSelector(submitButton, { timeout: 60000 })
 			await page.click(submitButton)
-
+			page.on("response", async response => {
+				try {
+					if (
+						response.request().method() === "POST" &&
+						response.url().endsWith("/virtual/tokenised") &&
+						response.headers()["content-type"].includes("application/json")
+					) {
+						if (response.status() != 201)
+							throw new Error(await response.json())
+					}
+				} catch (error) {
+					console.log(error)
+				}
+			})
```

Comment:
Does this work?
Is it not better to add this logic into the existing response listener, and add the listener at the top rather then after we've clicked?